packages/mattermost-redux/src/actions/posts.ts

```diff
@@ -776,21 +777,34 @@ export function getPosts(channelId: string, page = 0, perPage = Posts.POST_CHUNK
 
 export function getPostsUnread(channelId: string, fetchThreads = true, collapsedThreadsExtended = false) {
     return async (dispatch: DispatchFunc, getState: GetStateFunc) => {
+        const shouldLoadRecent = getUnreadScrollPositionPreference(getState()) === Preferences.UNREAD_SCROLL_POSITION_START_FROM_NEWEST;
         const collapsedThreadsEnabled = isCollapsedThreadsEnabled(getState());
         const userId = getCurrentUserId(getState());
         let posts;
+        let recentPosts;
         try {
             posts = await Client4.getPostsUnread(channelId, userId, DEFAULT_LIMIT_BEFORE, DEFAULT_LIMIT_AFTER, fetchThreads, collapsedThreadsEnabled, collapsedThreadsExtended);
+
+            if (posts.next_post_id && shouldLoadRecent) {
+                recentPosts = await Client4.getPosts(channelId, 0, Posts.POST_CHUNK_SIZE / 2, fetchThreads, collapsedThreadsEnabled, collapsedThreadsExtended);
```

Comment:
What's the motivation for loading two separate chunks of posts here instead of just loading the most recent posts? Is it so that we can still go back to the unread posts?

Reply:
>  Is it so that we can still go back to the unread posts?

Yes. I loaded two separate chunks to handle the logic related to "go back to the unread posts".